core/src/main/kotlin/org/evomaster/core/problem/external/service/ExternalService.kt

```diff
@@ -2,19 +2,59 @@ package org.evomaster.core.problem.external.service
 
 import com.github.tomakehurst.wiremock.WireMockServer
 
-class ExternalService (
+/**
+ * Represent the external service related information including
+ * WireMock server and ExternalServiceInfo collected from SUT.
+ */
+class ExternalService(
+    /**
+     * External service information collected from SUT
+     */
     val externalServiceInfo: ExternalServiceInfo,
+    /**
+     * Initiated WireMock server for the external service
+     */
     private val wireMockServer: WireMockServer
-        ) {
+) {
 
     /**
      * Return the IP address of WireMock instance
      */
-    fun getWireMockAddress() : String {
+    fun getWireMockAddress(): String {
         return wireMockServer.options.bindAddress()
     }
 
+    fun getWireMockServer(): WireMockServer {
+        return wireMockServer
+    }
+
+    /**
+     * To get all the HTTP/S requests made to the WireMock instance
+     */
+    fun getAllServedRequests(): MutableList<ExternalServiceRequest> {
```

Comment:
any specific reason to have a "mutable" list here?

Reply:
Nope. Had a different plan for this, when I wrote for the first time. Failed to change it. Fixed now.